packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManager.java

```diff
@@ -600,6 +601,7 @@ private static Layout createLayout(
             text.getSpanFlags(span));
         text.removeSpan(span);
       }
+      boring = BoringLayout.isBoring(text, paint);
```

Comment:
Quoting @javache:

> This is quite of an expensive call

Also we call this just above at the beginning of the `adjustSpannableFontToFit` function. Why do you need call this again?

Reply:
It updates the font metrics after applying the smaller font size. In this exact case, the layout height is larger than the available height and without the update to `boring` it ends up shrinking the text to the smallest allowed value:

<img width="348" alt="image" src="https://github.com/user-attachments/assets/02ee8dfb-303b-4ac5-b4b9-4c38103416ef">

The other two numbers end up as a `StaticLayout` so they aren't affected.

Reply:
We could probably guard this with a check to only recalculate BoringLayout if existing text was boring (adjusting font size won't change boringness, but will change the layout result). But I think needing to recreate it makes sense.

Reply:
Added a check in https://github.com/facebook/react-native/pull/47082/commits/fb3f1f0ac755007fcddf805a03cea02d878df2f2